### PR TITLE
libuldaq: init at 1.2.0

### PR DESCRIPTION
**Pull request by simonkampe**
###### Motivation for this change
libuldaq was missing in the package repository

###### Things done
Added maintainer simonkampe
Added libuldaq

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129469` at 2d790357 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>libuldaq</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/libraries/libuldaq/default.nix:13:15:
```
   |
13 |   patches = [ ./0001-uldaq.patch ];
   |               ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -12024,4 +12024,10 @@
     github = "zupo";
     githubId = 311580;
   };
+  simonkampe = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:12027]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/libraries/libuldaq/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, pkgs, stdenv, ...}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:2]:
```suggestion
{ lib, stdenv, fetchFromGitHub, autoreconfHook, libusb1 }:

stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/libuldaq/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "libuldaq";
+  version = "1.2.0";
+
+  src = pkgs.fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:6]:
```suggestion
  src = fetchFromGitHub {
```

File: pkgs/development/libraries/libuldaq/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "uldaq";
+    rev = "24ba881e81ceeb24ff924852c7f56dda072d0282";
+    sha256 = "0l9ima8ac99yd9vvjvdrmacm95ghv687wiy39zxm00cmghcfv3vj";
+  };
+
+  patches = [ ./0001-uldaq.patch ];
+
+  nativeBuildInputs = with pkgs; [ autoreconfHook ];
+
+  buildInputs = with pkgs; [ libusb1 ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:17]:
```suggestion
  nativeBuildInputs = [ autoreconfHook ];

  buildInputs = [ libusb1 ];
```

File: pkgs/development/libraries/libuldaq/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:19]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/libuldaq/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      Library used to communicate with USB data acquisition (DAQ)
+      devices from Measurement Computing
+    '';
+    homepage = "https://github.com/mccdaq/uldaq";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
+    maintainers = [ lib.maintainers.simonkampe ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:31]:
```suggestion
    license = licenses.mit;
    platforms = platforms.linux;
    maintainers = [ maintainers.simonkampe ];
  };
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (simonkampe) [reply maintainers/maintainer-list.nix:12027]: It already is a separate commit (3839fc0), should I put it in a separate PR? I will fix the sorting.

File: pkgs/development/libraries/libuldaq/default.nix (hunk 2 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed

File: pkgs/development/libraries/libuldaq/default.nix (hunk 3 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed

File: pkgs/development/libraries/libuldaq/default.nix (hunk 4 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed

File: pkgs/development/libraries/libuldaq/default.nix (hunk 5 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed

File: pkgs/development/libraries/libuldaq/default.nix (hunk 6 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix]: No, just fix the sorting.

File: pkgs/development/libraries/libuldaq/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      Library used to communicate with USB data acquisition (DAQ)
+      devices from Measurement Computing
+    '';
+    homepage = "https://github.com/mccdaq/uldaq";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.simonkampe ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:35]:
```suggestion
  };
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
please combine the review comments.

File: pkgs/development/libraries/libuldaq/default.nix (hunk 7 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed & amended

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (simonkampe) [reply maintainers/maintainer-list.nix]: Sorting is fixed.

**Comment by simonkampe** [comment]
> please combine the review comments.

I combined the comments on libuldaq/default.nix, should I squash both the sorting commit and the review changes commit into one also?

File: pkgs/development/libraries/libuldaq/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, libusb1, ...}:
+
+stdenv.mkDerivation rec {
+  pname = "libuldaq";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "mccdaq";
+    repo = "uldaq";
+    rev = "24ba881e81ceeb24ff924852c7f56dda072d0282";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libuldaq/default.nix:10]:
```suggestion
    rev = "v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please squash the package related commits.

File: pkgs/development/libraries/libuldaq/default.nix (hunk 8 above)
Comment (simonkampe) [reply pkgs/development/libraries/libuldaq/default.nix]: Fixed

**Comment by simonkampe** [comment]
> Also please squash the package related commits.

Commits now squashed into one for maintainer addition and one for package.
